Week 1/Contest (Mar 11)/Knot Knowledge/elijah.cs

```diff
@@ -6,10 +6,15 @@ static class App
     static void Main()
     {
         Console.ReadLine();
-        int[] needed = Console.ReadLine().Split(' ').Select(o => int.Parse(o)).ToArray();
-        int[] known = Console.ReadLine().Split(' ').Select(o => int.Parse(o)).ToArray();
-        int[] remaining = needed.Where(o => !known.Any(o2 => o2 == o)).ToArray();
         
-        Console.WriteLine(remaining[0]);
+        var needed = Console.ReadLine()
+            .Split(' ')
+            .Select(o => int.Parse(o));
+        
+        var known = Console.ReadLine()
+            .Split(' ')
+            .Select(o => int.Parse(o));
+        
+        Console.WriteLine(needed.First(o => !known.Any(o2 => o2 == o)));
```

Comment:
This is shorter indeed.

Challenge for you Elijah, can you make this faster also? :) what's the current time complexity?

Reply:
I believe this'll be O(n^2) on account of both the conditional First() and Any() both being O(n). I _might_ be able to get it down to O(log(n)), though I'll have to keep thinking on it.

Reply:
Yep, that is correct.

I think our goal here is to make it O(nlog(n)) or O(n)!